Open Judge System/Web/OJS.Web/Areas/Administration/Controllers/ContestsController.cs

```diff
@@ -537,6 +537,65 @@ public ActionResult TransferParticipants(ShortContestAdministrationViewModel mod
             return this.Redirect(returnUrl);
         }
 
+        [HttpGet]
+        public ActionResult CalculateContestLoad(int categoryId, int currentContestId)
+        {
+            if (!this.CheckIfUserHasContestPermissions(currentContestId))
+            {
+                return this.RedirectToContestsAdminPanelWithNoPrivilegesMessage();
+            }
+
+            var contests = this.contestsData
+                .GetAll()
+                .Where(c => c.CategoryId == categoryId && c.StartTime.HasValue && c.EndTime.HasValue)
+                .OrderByDescending(x => x.StartTime.Value)
+                .ToList();
+
+            var model = new ContestLoadCalculationViewModel();
+            foreach (var contest in contests)
+            {
+                if (contest.Id == currentContestId)
+                {
+                    model.ExamLengthInHours = (contest.EndTime.Value - contest.StartTime.Value).Hours;
+                    model.ExpectedExamProblemsCount = contest.ProblemGroups.Count();
+                    model.ExpectedStudentsCount = contest.Participants.Count();
+                    model.ContestName = contest.Name;
+                    model.AverageProblemRunTimeInSeconds = this.GetContestSubmissionsAverageRunTimeSeconds(contest);
+                    model.CurrentContestId = contest.Id;
```

Comment:
I'm not sure we want to calculate average run time for all the contests. We just need for the selected one.